client/packages/programs/src/JsonForms/common/components/Array/Notes.tsx

```diff
@@ -236,15 +239,23 @@ const NoteComponent = (props: ControlProps) => {
         multiline
         rows={rows}
         onChange={e => onChange(e.target.value)}
-        onBlur={() => setEditMode(false)}
+        onBlur={() => {
+          setEditMode(false);
+          if (text === undefined)
+            // Intentionally setting an invalid response so it can be
+            // distinguished from a completely empty (also invalid) response for
+            // the purposes of error checking
+            handleChange(path, {
+              text,
+              invalid: true,
+            });
```

Comment:
I don't really like putting bogus data into the JSON form data like this, but I couldn't find any other way for the parent to know what kind of "invalid" it is -- invalid because it's a new note, or because we've left it empty after input.